pkg/store/cache/caching_bucket_factory.go

```diff
@@ -60,5 +103,35 @@ func NewCachingBucketFromYaml(yamlContent []byte, bucket objstore.Bucket, logger
 		return nil, errors.Errorf("unsupported cache type: %s", config.Type)
 	}
 
-	return NewCachingBucket(bucket, c, config.CachingBucketConfig, logger, reg)
+	cb, err := NewCachingBucket(bucket, logger, reg)
+	if err != nil {
+		return nil, err
+	}
+
+	cbc := config.CachingBucketConfig
+	// Configure cache.
+	cb.CacheGetRange("chunks", c, isTSDBChunkFile, cbc.ChunkSubrangeSize, cbc.ChunkObjectSizeTTL, cbc.ChunkSubrangeTTL, cbc.MaxChunksGetRangeRequests)
+	cb.CacheExists("metafile", c, isMetaFile, cbc.MetafileExistsTTL, cbc.MetafileDoesntExistTTL)
+	cb.CacheGet("metafile", c, isMetaFile, cbc.MetafileContentTTL, cbc.MetafileExistsTTL, cbc.MetafileDoesntExistTTL)
+
+	// Cache Iter requests for root.
+	cb.CacheIter("dir", c, func(dir string) bool { return dir == "" }, cbc.RootIterTTL)
```

Comment:
I would define a standalone function for `func(dir string) bool { return dir == "" }` just for readability.